Formula/singular.rb

```diff
@@ -0,0 +1,34 @@
+class Singular < Formula
+  desc "Computer algebra system for polynomial computations"
+  homepage "https://www.singular.uni-kl.de/"
+  url "http://www.mathematik.uni-kl.de/ftp/pub/Math/Singular/src/4-1-0/singular-4.1.0p3.tar.gz"
+  version "4.1.0p3"
+  sha256 "440164c850d5a1575fcbfe95ab884088d03c0449570d40f465611932ffd0bf80"
+  head "https://github.com/Singular/Sources.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "gmp"
+  depends_on "ntl"
+
+  def install
+    system "./autogen.sh"
```

Comment:
The release tarball includes configure so we don't need to run autogen.sh

Reply:
But it is needed when installing with --HEAD. In this case autoconf, automake and libtool are needed too. Is there a way to make dependencies depend on install options?

Reply:
See `fontconfig` for an example.